File: frontend/lib/models/restaurant.dart
class Restaurant {
  final int id;
  final String name;
  final String description;
  final String phoneNumber;
  final String? logoUrl;
  final String? bannerUrl;
  final double deliveryFee;
  final double minimumOrder;
  final int deliveryTimeMin;
  final int deliveryTimeMax;
  final bool isOpen;
  final bool isActive;
  final double rating;
  final int totalReviews;
  final Address? address;
  final List<Category> categories;

  Restaurant({
    required this.id,
    required this.name,
    required this.description,
    required this.phoneNumber,
    this.logoUrl,
    this.bannerUrl,
    required this.deliveryFee,
    required this.minimumOrder,
    required this.deliveryTimeMin,
    required this.deliveryTimeMax,
    required this.isOpen,
    required this.isActive,
    required this.rating,
    required this.totalReviews,
    this.address,
    this.categories = const [],
  });

  factory Restaurant.fromJson(Map<String, dynamic> json) {
    return Restaurant(
      id: json['id'],
      name: json['name'],
      description: json['description'] ?? '',
      phoneNumber: json['phoneNumber'],
      logoUrl: json['logoUrl'],
      bannerUrl: json['bannerUrl'],
      deliveryFee: (json['deliveryFee'] as num).toDouble(),
      minimumOrder: (json['minimumOrder'] as num).toDouble(),
      deliveryTimeMin: json['deliveryTimeMin'],
      deliveryTimeMax: json['deliveryTimeMax'],
      isOpen: json['isOpen'],
      isActive: json['isActive'],
      rating: (json['rating'] as num).toDouble(),
      totalReviews: json['totalReviews'],
      address: json['address'] != null ? Address.fromJson(json['address']) : null,
      categories: json['categories'] != null
          ? (json['categories'] as List).map((cat) => Category.fromJson(cat)).toList()
          : [],
    );
  }

  String get deliveryTimeRange => '$deliveryTimeMin-${deliveryTimeMax}min';
  String get formattedRating => rating.toStringAsFixed(1);
}

class Address {
  final int id;
  final String street;
  final String number;
  final String? complement;
  final String neighborhood;
  final String city;
  final String state;
  final String zipCode;
  final double? latitude;
  final double? longitude;

  Address({
    required this.id,
    required this.street,
    required this.number,
    this.complement,
    required this.neighborhood,
    required this.city,
    required this.state,
    required this.zipCode,
    this.latitude,
    this.longitude,
  });

  factory Address.fromJson(Map<String, dynamic> json) {
    return Address(
      id: json['id'],
      street: json['street'],
      number: json['number'],
      complement: json['complement'],
      neighborhood: json['neighborhood'],
      city: json['city'],
      state: json['state'],
      zipCode: json['zipCode'],
      latitude: json['latitude']?.toDouble(),
      longitude: json['longitude']?.toDouble(),
    );
  }

  String get fullAddress {
    final complement = this.complement?.isNotEmpty == true ? ', ${this.complement}' : '';
    return '$street, $number$complement, $neighborhood, $city - $state, $zipCode';
  }
}

class Category {
  final int id;
  final String name;
  final String? description;
  final String? iconUrl;

  Category({
    required this.id,
    required this.name,
    this.description,
    this.iconUrl,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      iconUrl: json['iconUrl'],
    );
  }
}

File: backend/src/main/java/com/openbag/modules/restaurant/entity/Restaurant.java
package com.openbag.modules.restaurant.entity;

import com.openbag.modules.user.entity.User;
import com.openbag.modules.user.entity.Address;
import com.openbag.modules.organization.entity.Organization;
import com.openbag.modules.product.entity.Product;
import com.openbag.modules.product.entity.Category;
import com.openbag.modules.order.entity.Order;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "restaurants")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Restaurant {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    @Size(max = 100)
    private String name;

    @Size(max = 500)
    private String description;

    @NotBlank
    @Size(max = 15)
    @Column(name = "phone_number")
    private String phoneNumber;

    @NotBlank
    @Size(max = 18)
    @Column(unique = true)
    private String cnpj;

    @Column(name = "logo_url")
    private String logoUrl;

    @Column(name = "banner_url")
    private String bannerUrl;

    @NotNull
    @Column(name = "delivery_fee", precision = 10, scale = 2)
    private BigDecimal deliveryFee;

    @NotNull
    @Column(name = "minimum_order", precision = 10, scale = 2)
    private BigDecimal minimumOrder;

    @NotNull
    @Column(name = "delivery_time_min")
    private Integer deliveryTimeMin;

    @NotNull
    @Column(name = "delivery_time_max")
    private Integer deliveryTimeMax;

    @Column(precision = 10, scale = 7)
    private BigDecimal latitude;

    @Column(precision = 10, scale = 7)
    private BigDecimal longitude;

    @Column(name = "is_open")
    private boolean isOpen = true;

    @Column(name = "is_active")
    private boolean isActive = true;

    @Column(precision = 3, scale = 2)
    private BigDecimal rating = BigDecimal.ZERO;

    @Column(name = "total_reviews")
    private Integer totalReviews = 0;

    @CreationTimestamp
    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "owner_id")
    private User owner;

    // Organização à qual o restaurante pode pertencer (opcional)
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "organization_id")
    private Organization organization;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "address_id")
    private Address address;

    @OneToMany(mappedBy = "restaurant", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<Product> products = new ArrayList<>();

    @OneToMany(mappedBy = "restaurant", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<Order> orders = new ArrayList<>();

    @ManyToMany
    @JoinTable(
        name = "restaurant_categories",
        joinColumns = @JoinColumn(name = "restaurant_id"),
        inverseJoinColumns = @JoinColumn(name = "category_id")
    )
    private List<Category> categories = new ArrayList<>();

    @OneToOne(mappedBy = "restaurant", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private LayoutConfig layoutConfig;

    @OneToMany(mappedBy = "restaurant", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<OpeningHour> openingHours = new ArrayList<>();
}
